src/main/java/org/springframework/cloud/vault/UserIdMechanism.java

```diff
@@ -0,0 +1,171 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.vault;
+
+import java.io.IOException;
+import java.net.InetAddress;
+import java.net.NetworkInterface;
+import java.nio.charset.StandardCharsets;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.Collections;
+import java.util.List;
+
+import lombok.extern.apachecommons.CommonsLog;
+
+import org.springframework.security.crypto.codec.Hex;
+
+/**
+ * AppId authentication methods.
+ * 
+ * @author Mark Paluch
+ */
+@CommonsLog
+public enum UserIdMechanism {
```

Comment:
This seems a little strange to me to use an enum. Why not use an interface with multiple implementations that is activated by a property? It also needs to support an unauthenticated mode, no?